res/controllers/midi-components-0.0.js

```diff
@@ -677,8 +673,7 @@
                 this.deckNumbers = [deckNumbers];
             }
         } else {
-            print("ERROR! new Deck() called without specifying any deck numbers");
-            return;
+            throw Error("new Deck() called without specifying any deck numbers");
```

Comment:
This error is wrong.
`XoneK2.Deck.prototype = new components.Deck();`
which is correct, triggers the error, but nothing is actually wrong.

Reply:
I think the error is correct here. `this.currentDeck` and `this.deckNumbers` is undefined when you don't pass any deck numbers, which will result in errors when you call `toggle()`.

https://github.com/mixxxdj/mixxx/blob/0fe974af8fb86ff5f0c6dc11c955f29858e145d1/res/controllers/midi-components-0.0.js#L662-L714

Why are you even using a deck when you don't pass deckNumbers? Shouldn't you use `components.ComponentContainer` instead?

Reply:
JavaScript is weird. Setting the prototype of my XoneK2.Deck object to a constructed components.Deck object is required to inherit from it.

Reply:
> Setting the prototype of my XoneK2.Deck object to a constructed components.Deck object is required to inherit from it.

Wait, isn't the proper way to do it like this?

```js
XoneK2.Deck.prototype = Object.create(components.Deck.prototype);
```

Reply:
Does that not execute the constructor? JavaScript is bizarre.

Reply:
> Does that not execute the constructor? JavaScript is bizarre.

Thats the important difference between `Object.create` and `new`. From my understanding, both create a new object and set its prototype to the constructor in question. But `Object.create` doesn't execute the constructor, the constructor of the child class should do a `parent.call(this [, args])` in its constructor.

So IMO, Jan is actually correct. Unfortunately, `Òbject.create` was introduced in ES5, so we have to stay with `new` for inheritance and allow that the constructor is called without arguments.

In all honesty, I doubt this PR is really needed since we're gonna rewrite componentsJS anyways once at least #1795 is merged. Once we rewrote ComponentsJS (I estimate this will have happened in the next 6 months), developers should have stopped using components 0.0 anyways.

Reply:
Okay, I think we should close this without merging then.

Reply:
Agreed.